ledger/participant-integration-api/src/test/suite/scala/platform/indexer/parallel/BatchingParallelIngestionPipeSpec.scala

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.indexer.parallel
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.daml.ledger.api.testing.utils.AkkaBeforeAndAfterAll
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.duration.FiniteDuration
+
+class BatchingParallelIngestionPipeSpec
```

Comment:
Is it worth adding tests that check whether a slow ingestion leads to full batches?

Reply:
I don't think so. Firstly it wont necessarily lead to full batches. The groupedWithin will take whatever it has if the within part breached. Secondly: I think that would be rather testing the framework.